src/UCTSearch.cpp

```diff
@@ -235,6 +235,39 @@ void UCTSearch::dump_stats(FastState & state, UCTNode & parent) {
     tree_stats(parent);
 }
 
+void UCTSearch::output_analysis(FastState & state, UCTNode & parent) {
+    if (!parent.has_children()) {
+        return;
+    }
+
+    const int color = state.get_to_move();
+
+    // sort children, put best move on top
+    parent.sort_children(color);
```

Comment:
Not threadsafe to call during the search, see previous review. I think you can get rid of this, but you need to modify the loop below to not break early if there are no visits, just skip the current move (because moves will no longer be in visits or prior order).

Reply:
Ok, so the mutex lock in sort_children doesn't make it completely thread safe?

Reply:
The problem is that there is code that assumes it can iterate over m_children and probe or update any information that is in the atomic of the node structure (without taking locks), because the m_children array is not supposed to change once the node is expanded and/or the search is multi-threaded.

I just reviewed the code, and it looks like those users may be down to prune_noncontenders, actually. But it's good to keep the above invariant if we ever run into locking bottlenecking us. It's pretty annoying if you have to lock a node just to be able to gather statistics on the children.

Reply:
Uh, actually this function would itself be a user of that invariant too now. As long as the above invariant holds, it doesn't need to keep the root node locked while it's printing statistics. Which is good.

Reply:
Ok, so conclusion is don't sort.